Moodis/Moodis/Models/ImageInfo.cs

```diff
@@ -21,28 +23,37 @@ public struct Emotion
         public string ImagePath { get; set; }
         public Emotion[] emotions;
         private string id;
-        private string age;
-        private string gender;
+        private double? age;
+        private Gender? gender;
 
-        public void SetImageInfo(string jsonInfo)
+        public void SetImageInfo(IList<DetectedFace> faceList)
         {
-            dynamic data = JObject.Parse(jsonInfo);
-            id = data.faceId;
-            age = data.faceAttributes.age;
-            gender = data.faceAttributes.gender;
-            AddEmotions(data);
+            //TODO change to faceList.ForEach(face => action), when implementing identity feature
+            id = faceList[0].FaceId.ToString();
+            age = faceList[0].FaceAttributes.Age;
+            gender = faceList[0].FaceAttributes.Gender;
+            AddEmotions(faceList[0].FaceAttributes.Emotion);
         }
 
-        private void AddEmotions(dynamic data)
+        private void AddEmotions(Microsoft.Azure.CognitiveServices.Vision.Face.Models.Emotion detectedEmotions)
         {
             emotions = new Emotion[8];
-            int counter = 0;
-            foreach(dynamic emotion in data.faceAttributes.emotion)
-            {
-                emotions[counter].name = (string) emotion.Name;
-                emotions[counter].confidence = (double) emotion.Value;
-                counter++;
-            }
+            emotions[0].name = "Anger";
+            emotions[0].confidence = detectedEmotions.Anger;
+            emotions[1].name = "Contempt";
+            emotions[1].confidence = detectedEmotions.Contempt;
+            emotions[2].name = "Disgust";
+            emotions[2].confidence = detectedEmotions.Disgust;
+            emotions[3].name = "Fear";
+            emotions[3].confidence = detectedEmotions.Fear;
+            emotions[4].name = "Happiness";
+            emotions[4].confidence = detectedEmotions.Happiness;
+            emotions[5].name = "Neutral";
+            emotions[5].confidence = detectedEmotions.Neutral;
+            emotions[6].name = "Sadness";
+            emotions[6].confidence = detectedEmotions.Sadness;
+            emotions[7].name = "Neutral";
+            emotions[7].confidence = detectedEmotions.Surprise;
```

Comment:
Can't this be done in a foreach loop ?

Reply:
Change it if possible at least with simple for